pkg/utils/secrets/basic_auth.go

```diff
@@ -93,26 +111,52 @@ func (b *BasicAuth) SecretData() map[string][]byte {
 	case BasicAuthFormatNormal:
 		data[DataKeyUserName] = []byte(b.Username)
 		data[DataKeyPassword] = []byte(b.Password)
+
+		if b.BcryptPasswordHash != "" {
+			data[DataKeyPasswordHash] = []byte(b.BcryptPasswordHash)
+		}
+
 		fallthrough
 
 	case BasicAuthFormatCSV:
-		data[DataKeyCSV] = []byte(fmt.Sprintf("%s,%s,%s,%s", b.Password, b.Username, b.Username, user.SystemPrivilegedGroup))
+		var csvFormatStr bytes.Buffer
+
+		if b.BcryptPasswordHash != "" {
+			csvFormatStr.WriteString(b.BcryptPasswordHash + ",")
```

Comment:
I dug a bit about the `csv` format. It [looks like it is ](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#static-token-file) `token,user,uid,"group1,group2,group3"`.

So you should not add the bcrypt hash in the `csv`, I think.